src/main/java/soolutions/hometaskapp/common/Amount.java

```diff
@@ -8,26 +8,36 @@
   private final String currency;
 
   public Amount(double value, String currency) {
-    if (value < 0) {
-      throw new InvalidAmountException(value);
-    }
-
     this.value = value;
     this.currency = currency;
   }
 
+  public Amount zero() {
+      return new Amount(0, this.currency);
+  }
+
   public Amount add(Amount other) {
     if (!currency.equals(other.currency)) {
       throw new CurrencyMismatchException(
-          "Cannot add different currencies " + currency + " : " + other.currency);
+          "Cannot apply different currencies " + currency + " : " + other.currency);
     }
     return new Amount(this.value + other.value, currency);
   }
 
+  public Amount withdraw(Amount other) {
```

Comment:
Withdraw is an `Account` concept, not an `Amount` concept.

Reply:
Maybe it'll be better to use `subtract` from arithmetics?

Reply:
Yes.

Reply:
Done! Thanks.